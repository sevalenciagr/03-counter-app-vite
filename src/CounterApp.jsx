import { useState } from 'react'
import PropTypes from 'prop-types'

export const CounterApp = ( { value } ) => {

    const [ counter,setCounter ] = useState( value );

    const handleAdd = (event) => {
        setCounter( counter + 1 );
        // setCounter( (c) => c + 1 );
    };
    
    const handleSubstract = (event) => {
        setCounter( counter - 1 );
    };

    const handleReset = (event) => {
        setCounter( value );
    };


  return (
    <>
        <h1>CounterApp</h1>
        <h2> { counter } </h2>

        <button onClick={ handleAdd } >
            +1
        </button>
        <button onClick={ handleSubstract }>-1</button>
        <button onClick={ handleReset }>Reset</button>
    </>
  )
}

CounterApp.propTypes = {
    value: PropTypes.number
}
