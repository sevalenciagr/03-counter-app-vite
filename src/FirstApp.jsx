import PropTypes from "prop-types";

export const FirstApp = ( { title, subTitle } ) => {

  return (
    <>
        <h1>{ title }</h1>
        <p>{ subTitle }</p>
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string.isRequired
}

FirstApp.defaultProps = {
  name: 'Sebas',
  superTitle: 'No hay subtitulo',
  title: 'No hay titulo',
};