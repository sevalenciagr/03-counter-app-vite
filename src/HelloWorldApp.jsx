import React from 'react'

export const HelloWorldApp = () => {
  return (
    <h1>Hello World App</h1>
  )
}
