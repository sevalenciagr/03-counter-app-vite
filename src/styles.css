html, body {
    background-color: #2123A0;
    color: white;
    font-family: Helvetica, Arial, sans-serif;
    font-size: 1.3rem;
    padding: 70px;
}

button {
    padding: 15px;
    font-size: 1.3rem;
}