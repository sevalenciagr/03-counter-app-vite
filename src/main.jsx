import React from 'react'
import  ReactDOM  from "react-dom/client";
import { HelloWorldApp } from './HelloWorldApp';
import { FirstApp } from './FirstApp';
// import { CounterApp } from './CounterApp';
import './styles.css';



ReactDOM.createRoot( document.getElementById('root') ).render(
    <React.StrictMode>
        {/* <HelloWorldApp /> */}
        <FirstApp title='Prueba proptypes' subTitle='1234' /> 
        {/* <CounterApp value = { 0 } />    */}

    </React.StrictMode>
);