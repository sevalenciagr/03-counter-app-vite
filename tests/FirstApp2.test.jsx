import {  render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('Pruebas en <FirstApp />', () => {

    const title = 'Prueba proptypes';
    const subTitle = '1234';
    
    test('Debe hacer match con el snapshot', () => { 
        const { container } = render( <FirstApp title = { title } />);
        expect( container ).toMatchSnapshot();

     });

     test('Debe mostrar el mensaje "1234" ', () => { 

        render( <FirstApp title= { title }/>);
        expect( screen.getByText(title) ).toBeTruthy();

      });

      test('Debe mostrar el titulo en un h1', () => { 
       
        render( <FirstApp title= { title }/>);
        expect( screen.getByRole('heading', {level: 1}).innerHTML).toContain( title );

       })

       test('Debe mostrar el subtitulo enviado por props', () => { 

            render( 
                <FirstApp 
                    title={ title } 
                    subTitle={ subTitle }
                /> 
            );

            expect( screen.getAllByAltText( subTitle ).length ).toBe( 2 );

        })

});