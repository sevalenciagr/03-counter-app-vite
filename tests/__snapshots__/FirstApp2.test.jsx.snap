// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> Debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Prueba proptypes
  </h1>
  <p />
  <p />
  <p>
    Sebas
  </p>
</div>
`;
